packages/experimental/Menu/src/MenuPopover/useMenuPopover.ts

```diff
@@ -13,10 +13,23 @@ export const useMenuPopover = (_props: MenuPopoverProps): MenuPopoverState => {
   const dismissBehaviors = context.isControlled
     ? (['preventDismissOnKeyDown', 'preventDismissOnClickOutside'] as DismissBehaviors[])
     : undefined;
+  const directionalHint = getDirectionalHint(context.isSubmenu, I18nManager.isRTL);
 
   // Initial focus behavior differs per platform, Windows platforms move focus
   // automatically onto first element of Callout
   const setInitialFocus = Platform.OS === ('win32' as any) || Platform.OS === 'windows';
 
-  return { triggerRef, onDismiss, dismissBehaviors, setInitialFocus };
+  return { triggerRef, onDismiss, directionalHint, dismissBehaviors, setInitialFocus };
+};
+
+const getDirectionalHint = (isSubmenu: boolean, isRtl: boolean): DirectionalHint | undefined => {
+  if (!isSubmenu) {
+    return undefined;
+  }
+
+  if (isRtl) {
+    return 'leftTopEdge';
+  }
+
+  return 'rightTopEdge';
```

Comment:
I might have hardcoded this switch in native code for macOS Callout. I'll have to double check these two don't conflict with each other.